### PR TITLE
DENG-4426 Add Ray Serve template

### PR DESCRIPTION
**Pull request by abhi-agg**
Fixes https://mozilla-hub.atlassian.net/browse/DENG-4426

 - Added a template ray serve app (a python script)
 - Added requirements file, Dockerfile, serve config file for the app
 - Updated README to add instructions for how to deploy ray serve apps to production

### Goal(s) of this Pull Request:

-

### Example of how it works:

...

### Acceptance criteria:

In order to approve, reviewer must be able to...

- [x] ...

### CODE REVIEW

File: src/mozmlops/templates/README.md [hunk 1]
```diff
@@ -78,3 +78,81 @@ We know this workaround fails to account for scheduling and scripts: we're worki
...
+#### Create Dockerfile for your Ray Serve app
+Creating docker images is the [recommended way](https://mozilla-hub.atlassian.net/wiki/spaces/DATA/pages/785514640/Deploy+inference+servers+to+production+GKE+using+Ray+Serve#Containerization-of-Ray-Serve-application) to deploy Ray Serve apps to production.
+
+Create a [Dockerfile](https://docs.docker.com/reference/dockerfile/) for your Ray Serve app and add it to the "application repo"
+
+> [!IMPORTANT]
+> The "deployment repo" will build docker image using the Dockerfile that you add to your "application repo".
+
+#### Create a Serve config for your Ray Serve app
+The Serve config is the [recommended way](https://docs.ray.io/en/latest/serve/configure-serve-deployment.html#specify-parameters-through-the-serve-config-file) to deploy and update Ray Serve apps in production.
```
Comment (chelseatroy) [inline src/mozmlops/templates/README.md:116]: A link as to _why_ this is the recommended practice would be nice from my perspective :)

File: src/mozmlops/templates/README.md [hunk 2]
```diff
@@ -78,3 +78,81 @@ We know this workaround fails to account for scheduling and scripts: we're worki
 > [!NOTE]  
 > We have changed `offline` here to false: hat means we _do_ want our flow to integrate with Weights and Biases!
 
+## Running Inference Servers in Production
+
+Deploying your Ray Serve app to production requires changes in 3 different repositories: “application repo“, “function specific repo” and “deployment repo“ (please see [glossary](https://mozilla-hub.atlassian.net/wiki/spaces/DATA/pages/785514640/Deploy+inference+servers+to+production+GKE+using+Ray+Serve#Glossary) for details on these repositories).
```
Comment (chelseatroy) [inline src/mozmlops/templates/README.md:83]: I think it's worth mentioning up front that, to get started and run an app _locally_, they only need to be in the application repo. It's daunting to think 3 repos will be needed up front; once the thing is working locally, _then_ they can make changes in the other two in order to permit deployment.

File: src/mozmlops/templates/requirements.rayserve.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Common dependencies to run any Ray Serve app
+ray[serve]==2.9.3
+
+# Dependencies specific to the Ray Serve app (template Ray Serve app of this repository in this case)
```
Comment (chelseatroy) [inline src/mozmlops/templates/requirements.rayserve.txt:4]: I might go with "_this_ ray serve app"

File: src/mozmlops/templates/serve_config.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# This file was generated using the `serve build` command on Ray v2.9.3.
+# It was further tweaked to adapt it to the template Ray Serve app in this repository.
```
Comment (chelseatroy) [inline src/mozmlops/templates/serve_config.yaml:2]: Might be worth putting a comment on the lines where it was tweaked, explaining why they were tweaked, to illustrate what we might mean by an instruction like "auto-generate and then tweak to your requirments."

File: src/mozmlops/templates/template_ray_serve.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+from transformers import pipeline
+
+app = FastAPI()
+
+
+class TranslateRequest(BaseModel):
+    text: str = Field(description="The text to translate")
+
+
+@serve.deployment()
```
Comment (chelseatroy) [inline src/mozmlops/templates/template_ray_serve.py:21]: Tiny! Simple! Perfect! 🤩

File: src/mozmlops/templates/README.md [hunk 6]
```diff
@@ -78,3 +78,81 @@ We know this workaround fails to account for scheduling and scripts: we're worki
...
+Create a [Dockerfile](https://docs.docker.com/reference/dockerfile/) for your Ray Serve app and add it to the "application repo"
+
+> [!IMPORTANT]
+> The "deployment repo" will build docker image using the Dockerfile that you add to your "application repo".
+
+#### Create a Serve config for your Ray Serve app
+The Serve config is the [recommended way](https://docs.ray.io/en/latest/serve/configure-serve-deployment.html#specify-parameters-through-the-serve-config-file) to deploy and update Ray Serve apps in production.
+
+1. Auto-generate the Serve config file using [serve build](https://docs.ray.io/en/latest/serve/api/index.html#serve-build) command
+
```
Comment (chelseatroy) [inline src/mozmlops/templates/README.md:119]: I love the specificity of all these steps!!! 🤩

**Review by chelseatroy — COMMENTED** [review]
Comments added! Mostly related to clarity and messaging for folks doing this for the first time. Will stand by for re-review :)

File: src/mozmlops/templates/README.md (hunk 1 above)
Comment (abhi-agg) [reply src/mozmlops/templates/README.md:116]: @chelseatroy I am changing this link to another link where the explanation for _why_ is also included. Thanks for the suggestion :)

File: src/mozmlops/templates/README.md (hunk 2 above)
Comment (abhi-agg) [reply src/mozmlops/templates/README.md:83]: I am adding a statement right after the current statement to clarify this.

File: src/mozmlops/templates/serve_config.yaml (hunk 4 above)
Comment (abhi-agg) [reply src/mozmlops/templates/serve_config.yaml]: Done.

**Comment by abhi-agg** [comment]
@chelseatroy Thanks for reviewing the PR. I addressed all the comments. It is ready for a re-review 👍🏾

**Review by chelseatroy — APPROVED** [review]
I think this looks good! Great work!

**Pull request merged**